src/unit_test.c

```diff
@@ -0,0 +1,332 @@
+/***********************************************************************
+ * Copyright (c) 2025  Matias Furszyfer (furszy)                       *
+ * Distributed under the MIT software license, see the accompanying    *
+ * file COPYING or https://www.opensource.org/licenses/mit-license.php.*
+ ***********************************************************************/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#if defined(SUPPORTS_CONCURRENCY) && SUPPORTS_CONCURRENCY
```

Comment:
nit: is the `defined(SUPPORTS_CONCURRENCY)` part needed? it's not used in similar pre-processor conditions below

Reply:
Done, updated.

Reply:
If I've grepped the codebase correctly, this will be the only external definition used as `#if DEFINITION`.

> The standard mandates that undefined macros evaluate to 0 in an `#if`.

Nevertheless, this approach has proven fragile. I would kindly ask to reconsider the design and use `#if defined(DEFINITION)` instead, which would also require slight adjustment in both build systems.

Reply:
Based on the CI response, it seems Windows dislikes undefined values.